app/controllers/api/canvas_proxy_controller.rb

```diff
@@ -6,20 +6,23 @@ class Api::CanvasProxyController < Api::ApiApplicationController
 
   def proxy
     result = canvas_api.proxy(params[:lms_proxy_call_type], params.to_unsafe_h, request.body.read, params[:get_all])
-    response.status = result.code
-
     allowed_headers = %w{
       content-type link p3p x-canvas-meta x-canvas-user-id
       x-rate-limit-remaining x-request-context-id x-request-cost
       x-request-processor x-robots-tag x-runtime x-session-id
       x-ua-compatible x-xss-protection
     }
 
-    result.headers.each do |name, val|
-      response.headers[name] = val if allowed_headers.include?(name)
+    if result.class == Array
```

Comment:
Also because the LMS API gem does not give us back the headers when it handles pagination. This code already exists in act, but should be here as well. I agree though that it would be good to refactor the gem so we can actually return the correct status codes and things.